public/app/services/draftsService.js

```diff
@@ -0,0 +1,59 @@
+angular.module('kibibitCodeEditor')
+
+.service('DraftsService', [
+  '$q',
+  'DRAFTS',
+  'SessionStorageService',
+  function(
+    $q,
+    DRAFTS,
+    SessionStorageService) {
+    var vm = this;
+
+    vm.draftsObject = {};
+
+    vm.deleteDraft = deleteDraft;
+    vm.getDraft = getDraft;
+    vm.saveDraft = saveDraft;
+
+    init();
+
+    ////////////
+
+    function deleteDraft(filePath) {
+      var objectKey = DRAFTS.DRAFT_PREFIX + filePath;
+      vm.draftsObject[objectKey] = undefined;
+      SessionStorageService.removeItem(objectKey);
+    }
+
+    function getDraft(filePath) {
+      var requestedFile = vm.draftsObject[filePath];
+      if (requestedFile) {
+        return $q.when(requestedFile);
```

Comment:
not sure that I like that syntax since I can't figure out why it's called `$q.when`

I personally prefer this syntax:

```javascript
function getDraft(filePath) {
  // There will always be a promise so always declare it.
  var deferred = $q.defer();
  var requestedFile = vm.draftsObject[filePath];
  if (requestedFile) {
    // Resolve the deferred $q object before returning the promise
    deferred.resolve(requestedFile);
  } else {
    // Reject the deferred $q object before returning the promise
    deferred.reject('requested file is not in sessionStorage');
  }
  // Return the promise. Since it was already resolved or rejected, this will be resolved immediatly
  return deferred.promise;
}
```

ofc, without the comments. the comments are for you 😉